File: samples/ObjC Collection View Inbox/Sample/MainVC.m
/*
 * Licensed Materials - Property of IBM
 *
 * 5725E28, 5725I03
 *
 * © Copyright IBM Corp. 2011, 2017
 * US Government Users Restricted Rights - Use, duplication or disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
 */
#import "MainVC.h"
#import <IBMMobilePush/IBMMobilePush.h>

@interface MainVC ()
@property (nonatomic, strong) id previewingContext;
@end

@implementation MainVC

-(void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    [self inboxUpdate];
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    [version setText: MCESdk.sharedInstance.sdkVersion];
    
    if ([self isForceTouchAvailable]) {
        self.previewingContext = [self registerForPreviewingWithDelegate:self sourceView:self.view];
    }
    
    // Show Inbox counts on main page
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(inboxUpdate) name:@"MCESyncDatabase" object:nil];
    if(MCERegistrationDetails.sharedInstance.mceRegistered)
    {
        [[MCEInboxQueueManager sharedInstance] syncInbox];
    }
    else
    {
        [NSNotificationCenter.defaultCenter addObserverForName: MCERegisteredNotification object:nil queue:NSOperationQueue.mainQueue usingBlock:^(NSNotification * _Nonnull note) {
            [[MCEInboxQueueManager sharedInstance] syncInbox];
        }];
    }
}

// Show Inbox counts on main page
-(void)inboxUpdate
{
    NSArray * messages = [[MCEInboxDatabase sharedInstance] inboxMessagesAscending: true];
    int unreadCount = 0;
    for(MCEInboxMessage * message in messages)
    {
        if(!message.isRead)
        {
            unreadCount += 1;
        }
    }
    
    dispatch_async(dispatch_get_main_queue(), ^{
        NSString * subtitle = @"";
        if(MCERegistrationDetails.sharedInstance.mceRegistered)
        {
            subtitle = [NSString stringWithFormat:@"%lu messages, %d unread", messages.count, unreadCount];
        }
        inboxCell.detailTextLabel.text = subtitle;
        altInboxCell.detailTextLabel.text = subtitle;
        [self.tableView reloadData];
    });
}

- (BOOL)isForceTouchAvailable
{
    if ([self.traitCollection respondsToSelector:@selector(forceTouchCapability)]) {
        return self.traitCollection.forceTouchCapability == UIForceTouchCapabilityAvailable;
    }
    return NO;
}

- (UIViewController *)previewingContext:(id<UIViewControllerPreviewing>)previewingContext viewControllerForLocation:(CGPoint)location{
    
    CGPoint cellPostion = [self.tableView convertPoint:location fromView:self.view];
    NSIndexPath *path = [self.tableView indexPathForRowAtPoint:cellPostion];
    
    if (path) {
        UITableViewCell *tableCell = [self.tableView cellForRowAtIndexPath:path];
        UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"Main" bundle:nil];
        
        if(tableCell.restorationIdentifier)
        {
            UIViewController *previewController = [storyboard instantiateViewControllerWithIdentifier: tableCell.restorationIdentifier];
            if(previewController)
            {
                previewingContext.sourceRect = [self.view convertRect:tableCell.frame fromView:self.tableView];
                return previewController;
            }
        }
    }
    return nil;
}

-(void)previewingContext:(id<UIViewControllerPreviewing>)previewingContext commitViewController:(UIViewController *)viewControllerToCommit
{
    [self.navigationController pushViewController:viewControllerToCommit animated:TRUE];
}

@end

File: samples/SwiftSample/MainVC.swift
/*
 * Licensed Materials - Property of IBM
 *
 * 5725E28, 5725I03
 *
 * © Copyright IBM Corp. 2015, 2017
 * US Government Users Restricted Rights - Use, duplication or disclosure restricted by GSA ADP Schedule Contract with IBM Corp.
 */


import UIKit

class MainVC : UITableViewController, UIViewControllerPreviewingDelegate
{
    @IBOutlet weak var version: UILabel?
    var previewingContext: UIViewControllerPreviewing?
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        version!.text = MCESdk.sharedInstance().sdkVersion()
        if #available(iOS 9.0, *) {
            self.previewingContext = self.registerForPreviewing(with: self, sourceView: self.view)
        }
    }
    
    @available(iOS 9.0, *)
    func isForceTouchAvailable() -> Bool
    {
        if self.traitCollection.responds(to: #selector(getter: UITraitCollection.forceTouchCapability))
        {
            return self.traitCollection.forceTouchCapability == UIForceTouchCapability.available
        }
        return false
    }
    
    func previewingContext(_ previewingContext: UIViewControllerPreviewing, commit viewControllerToCommit: UIViewController) {
        self.navigationController?.pushViewController(viewControllerToCommit, animated: true)
    }
    
    @available(iOS 9.0, *)
    func previewingContext(_ previewingContext: UIViewControllerPreviewing, viewControllerForLocation location: CGPoint) -> UIViewController? {
        let cellPosition = self.tableView.convert(location, from: self.view)
        if let path = self.tableView.indexPathForRow(at: cellPosition)
        {
            if let tableCell = self.tableView.cellForRow(at: path)
            {
                let storyboard = UIStoryboard.init(name: "Main", bundle: nil)
                if let ident = tableCell.restorationIdentifier
                {
                    let previewController = storyboard.instantiateViewController(withIdentifier: ident)
                    previewingContext.sourceRect = self.view.convert(tableCell.frame, from: self.tableView)
                    return previewController
                }
            }
        }
        return nil
    }
}